packages/bundler-plugin-core/src/sentry/telemetry.ts

```diff
@@ -119,29 +127,61 @@ export function addPluginOptionTags(options: InternalOptions, hub: Hub) {
   }
 
   hub.setTag("node", process.version);
+
+  hub.setTags({
+    organization: org,
+    project,
+    bundler,
+  });
+
+  hub.setUser({ id: org });
 }
 
-/**
- * Makes a call to SentryCLI to get the Sentry server URL the CLI uses.
- *
- * We need to check and decide to use telemetry based on the CLI's respone to this call
- * because only at this time we checked a possibly existing .sentryclirc file. This file
- * could point to another URL than the default URL.
- */
-export async function turnOffTelemetryForSelfHostedSentry(cli: SentryCLILike, hub: Hub) {
-  const cliInfo = await cli.execute(["info"], false);
+export async function shouldSendTelemetry(options: InternalOptions): Promise<boolean> {
+  const { silent, org, project, authToken, url, vcsRemote, customHeader, dist, telemetry } =
+    options;
 
-  const url = cliInfo
-    ?.split(/(\r\n|\n|\r)/)[0]
+  // `options.telemetry` defaults to true
+  if (telemetry === false) {
+    return false;
+  }
```

Comment:
As discussed, let's also check that `dryRun` isn't isn't enabled (or return `false` if it is)